subprojects/logging/src/integTest/groovy/org/gradle/DeprecationHandlingIntegrationTest.groovy

```diff
@@ -45,9 +80,9 @@ class DeprecatedTask extends DefaultTask {
         run()
 
         then:
-        output.contains("Build file '$buildFile': line 3")
+        output.contains("Build file '$buildFile': line 10")
```

Comment:
Actually, line 3 and line 6 should still be reported as the first calls to a deprecated method. Regarding the stacktrace it seems to me that the call to `nagUserOfDiscontinuedMethod` should never be reported in the message but what we want to report is the call to the method calling `nagUserOfDiscontinuedMethod`. This would actually mean to always drop the first (or last depending on your point of view) Stack frame and then reporting the next. This should happen while keeping the special handling for build files.